src/libsystemd/sd-device/device-util.c

```
@@ -113,3 +114,83 @@ bool device_match_parent(sd_device *device, Set *match_parent, Set *nomatch_pare
 
         return false;
 }
+
+static int add_string_field(
+                sd_device *device,
+                const char *field,
+                int (*func)(sd_device *dev, const char **s),
+                char ***ret) {
+
+        const char *s;
+        int r;
+
+        assert(device);
+        assert(field);
+        assert(func);
+
+        r = func(device, &s);
+        if (r < 0 && r != -ENOENT)
+                log_error_errno(r, "Failed to get device \"%s\" property: %m", field);
```

Please downgrade error and mention that the failure is ignored.